lib/browser/api/utility-process.ts

```diff
@@ -133,10 +106,10 @@ export default class UtilityProcess extends EventEmitter {
         }
         return false;
       } else if (channel === 'stdout' && this.#stdout) {
-        this.#stdout.storeInternalData(Buffer.from(args[1]), args[2]);
+        new Socket({ fd: args[0], readable: true }).pipe(this.#stdout);
```

Comment:
surprising/cool that this works :O

Reply:
There was a bit of learning involved for me here, I first used `CreateReadStream` but that ended up always hanging the main process on exit. When libuv performs the blocking read operation on a separate thread, the main thread waits for the reader thread to yield but never happens if the child process exited while the read operation was started. Turns out the recommended way to do this is via a `net.Socket` which actually performs a polling based read and it is what Node.js also uses internally https://github.com/nodejs/node/blob/1124558b655c2752bdb1a0bda99195b55b5d63ee/lib/internal/child_process.js#L334-L336

Reply:
cool. +1 for reusing Node.js's primitives.